benchmarks/src/test/java/org/apache/druid/benchmark/frame/FrameChannelMergerBenchmark.java

```diff
@@ -0,0 +1,353 @@
+
+  @Param({"20"})
+  private int keyLength;
+
+  @Param({"100"})
+  private int rowLength;
+
+  /**
+   * Linked to {@link KeyGenerator}.
+   */
+  @Param({"random", "sequential"})
+  private String keyGeneratorString;
+
+  /**
+   * Linked to {@link ChannelDistribution}.
+   */
+  @Param({"round_robin", "clustered"})
+  private String channelDistributionString;
+
+  /**
+   * Generator of keys.
+   */
+  enum KeyGenerator
+  {
+    /**
+     * Random characters from a-z.
+     */
+    RANDOM {
+      @Override
+      public String generateKey(int rowNumber, int keyLength)
+      {
+        final StringBuilder builder = new StringBuilder(keyLength);
+        for (int i = 0; i < keyLength; i++) {
+          builder.append((char) ('a' + ThreadLocalRandom.current().nextInt(26)));
+        }
+        return builder.toString();
+      }
+    },
+
+    /**
+     * Sequential with zero-padding.
+     */
+    SEQUENTIAL {
+      @Override
+      public String generateKey(int rowNumber, int keyLength)
+      {
+        return StringUtils.format("%0" + keyLength + "d", rowNumber);
+      }
+    };
+
+    public abstract String generateKey(int rowNumber, int keyLength);
+  }
+
+  /**
+   * Distribution of rows across channels.
+   */
+  enum ChannelDistribution
+  {
+    /**
+     * Sequential keys are distributed round-robin to channels.
+     */
+    ROUND_ROBIN {
+      @Override
+      public int getChannelNumber(int rowNumber, int numRows, int numChannels)
+      {
+        return rowNumber % numChannels;
+      }
+    },
+
+    /**
+     * Sequential keys are clustered into the same channels.
+     */
+    CLUSTERED {
+      @Override
+      public int getChannelNumber(int rowNumber, int numRows, int numChannels)
+      {
+        final int rowsPerChannel = numRows / numChannels;
+        return rowNumber / rowsPerChannel;
+      }
+    };
+
+    public abstract int getChannelNumber(int rowNumber, int numRows, int numChannels);
+  }
+
+  private final RowSignature signature =
+      RowSignature.builder()
+                  .add(KEY, ColumnType.STRING)
+                  .add(VALUE, ColumnType.STRING)
+                  .build();
+
+  private final FrameReader frameReader = FrameReader.create(signature);
+  private final List<KeyColumn> sortKey = ImmutableList.of(new KeyColumn(KEY, KeyOrder.ASCENDING));
+
+  private List<List<Frame>> channelFrames;
+  private FrameProcessorExecutor exec;
+  private List<BlockingQueueFrameChannel> channels;
+
+  /**
+   * Create {@link #numChannels} channels in {@link #channels}, with {@link #numRows} total rows split across the
+   * channels according to {@link ChannelDistribution}. Each channel is individually sorted, as required
+   * by {@link FrameChannelMerger}.
+   *
+   * Rows are fixed-length at {@link #rowLength} with fixed-length keys at {@link #keyLength}. Keys are generated
+   * by {@link KeyGenerator}.
+   */
+  @Setup(Level.Trial)
+  public void setupTrial() throws IOException
```

Comment:
```suggestion
  public void setupTrial()
```

Reply:
It seems no need to throw IOException here :sweat_smile:

Reply:
yep, fixed.